### ISSUE TITLE
What happend with chroot 3.15 and Nexus 7 2012 Wifi (grouper)

### ISSUE THREAD

**Issue by pindonga1234** [offensive-security/kali-nethunter#630]

**Comment by jcadduono** [comment]
nexus 2012 discontinued support due to kernel version 3.1 (kali rolling and newer requires kernel 3.2+, preferably 3.4+)

you will have to find the old sana chroot and use that for now, you can set your sources to use old.kali

**Comment by akrolla** [comment]
Try anykernel for nexus 7 I've had no issues using it except when I
installed kernel in nethunter package and it slows my nexus5 to a
screeching halt. So to mitigate I install kernel seperately and then do a
NH install

**Comment by pindonga1234** [comment]
Thank akrolla, but I dont have the more minimal idea how I have to do for install a kernel seperately. If you wanna help me with this I would thank. Jcadduono I searched on google old.kali and add this repositories
The Kali Rolling Repository

kali-rolling is our current active repository since the release of Kali 2016.1. Kali Rolling users are expected to have the following entries in their sources.list:
deb http://http.kali.org/kali kali-rolling main contrib non-free
# For source package access, uncomment the following line
# deb-src http://http.kali.org/kali kali-rolling main contrib non-free
Retired Kali sana (2.0) Repositories

For access to the retired sana repositories, have the following entries in your sources.list:
deb http://old.kali.org/kali sana main non-free contrib
# For source package access, uncomment the following line
# deb-src http://old.kali.org/kali sana main non-free contrib
But I received a lot of error. I know it is a big work but can you help me and give the lines to add for sources.list?. Thank

**Comment by jcadduono** [comment]
`deb http://old.kali.org/kali sana main non-free contrib`  
is all you need

**Comment by akrolla** [comment]
According to jcadduono the issue seems to be fixed just try to reinstall
kernel first in twrp and then nethunter 2nd but @jcadduono he said the
installer has a new kernel in it so i suggest download from either the
github or kali.org factoey wipe, re root install busybox and twrp annd
transfer the nethunter zip to downloads om the phone and in twrp recovery
install the NH zip

**Comment by pindonga1234** [comment]
Thank I will try

**Comment by jcadduono** [comment]
No it's not something that will be fixed, grouper is discontinued and new chroot cannot be used on it due to 3.1 kernel, there is no newer kernel compatible with it.

You must manually install sana chroot using app, and use sana old sources.list

**Comment by pindonga1234** [comment]
When I wanna install a package, say me that I need update my kernel, but I dont know how I can do this. Can you help me?. Im a lammers and I know is so hard explain me, and I know you dont have to time enough for help me only me.

**Comment by akrolla** [comment]
Have to find a compatible 3.4 kernel look at the othwr mexus 7 issue I opened

**Comment by pindonga1234** [comment]
Thank a lot akrolla, I red your very useful post and now I can resolved the issues. I have to pay some beers ha ha ha. In my country is the best way to thank a favor

**Comment by compguymn** [comment]
Hey, Im currently having the same problem 2012 nexus 7 grouper lollipop, kali nethunter 3.0.
 changing repos manually didnt work. What kernal did you flash over to resolve this issue. is there a good link or site if you share. Any help is much appreciated.

**Comment by akrolla** [comment]
find a 3.4 kernel that has injection capabilities. i somehow fooled my grouper into thinkin it was tilapia with a tilapia rom, and added the 3.4 kernel, or you could try the nethunter tilapia package , im not sure if it works never tried.

**Comment by jcadduono** [comment]
grouper and tilapia are the same device, they both use the same kernels, the only difference is that the kernel detects LTE support on the lte model, and ignores the lte drivers on the wifi model

**Comment by akrolla** [comment]
so is there a nethunter tilapia?

**Comment by jcadduono** [comment]
no grouper is discontinued

**Comment by akrolla** [comment]
so is tilapia discontinued?

**Comment by akrolla** [comment]
i know its just grrouper with phone capabilities

**Comment by akrolla** [comment]
is there a tilapia nh rom?

**Comment by akrolla** [comment]
ok so I guess inorder to do this back to the Frankenstein,  find a 3.4
kernel that allows for packet manipulation, download nh and nhterm apk and
symlink ..hopefully sounds like a tall order-_-

**Comment by 0E800** [comment]
I'll be releasing a alpha build of Nethunter for Oreo for Nexus 7 2013 WiFi aka Razor Flo.

Features AICP Oreo and ElementalX 7  w/ proper Keyboard gadget / HID and packet injection for TP Link adapter. 

Not working : prevent device from rebooting when external wifi adapter is removed .

I am working on Nexus 7 2012 Grouper and Tilapia.

Looking for Deb testers as I only have a Flo.


![Imgur](http://i.imgur.com/IQAjoU3.jpg)

**Comment by mronikoyi** [comment]
> @akrolla
> I'll be releasing a alpha build of Nethunter for Oreo for Nexus 7 2013 WiFi aka Razor Flo.
> 
> Features AICP Oreo and ElementalX 7 w/ proper Keyboard gadget / HID and packet injection for TP Link adapter.
> 
> Not working : prevent device from rebooting when external wifi adapter is removed .
> 
> I am working on Nexus 7 2012 Grouper and Tilapia.
> 
> Looking for Deb testers as I only have a Flo.
> 
> ![Imgur](https://camo.githubusercontent.com/4caecea5a0332326b74784d84561107cfb8e243e/687474703a2f2f692e696d6775722e636f6d2f4951416a6f55332e6a7067)

So, have you managed to get thngs working for Tilapia?

I actually have both Tilapia and Grouper.
Evrytime I try putting AOSP image and then adding NH image, I end up bricking my tablets.
Works well when I stay on kernel 3.1.1 but then i can't upgrade my packages because i need 3.2+ kernel.

I'm feeling like i'm going crazy!

**Comment by nikmel420** [comment]
Ive got 2 groupers that i have been trying to get to work forever. Lol i missed the memo on the kernal. After work gonna look at the sanza repository mentioned here. Glad i stumbled on this thread. And for the guy looking for testers. It would be a pleasure if youd include me. Im not sure whats on them now i havent touched them in at least a year. I gotta try the flashimg kernel seperate on my hamerheads also. they do crawl. But work perfect

**Comment by 0E800** [comment]
Apologies, life interefered again and I got sidetracked.

But I did add a Nexus 5 and a PixelXL to my collection and will try to get those working as well with Nethunter.

I see there are 2 Magisk modules for Nethunter. One add the neccesary wifi firmware support and the other handles the Kali chroot environment. Its possible that all that is needed is a kernel with the needed modules for injection and HID keyboard.
